test/cases/api.sh

```diff
@@ -0,0 +1,169 @@
+
+CurlCmd='curl -s -w %{http_code}'
+Header="x-rh-identity: $AuthString0"
+Address="localhost"
+Port="8086"
+Version="1.0"
+MajorVersion="1"
+BaseUrl="http://$Address:$Port/api/image-builder/v$Version"
+BaseUrlMajorVersion="http://$Address:$Port/api/image-builder/v$MajorVersion"
+
+function GetResponse() {
+  arr=($1)
+  echo "${arr[@]::${#arr[@]}-1}"
+}
+
+function GetExitCode() {
+  arr=($1)
+  echo "${arr[-1]}"
+}
+
+### Case: Get version
+function Test_GetVersion() {
+  result=$($CurlCmd -H "$Header" "$url/version")
+  ver="$(GetResponse "$result" | jq .version -r)"
+  [[ $ver == "$Version" ]]
+  exit_code=$(GetExitCode "$result")
+  [[ $exit_code == 200 ]]
+}
+
+### Case: Get openapi.json
+function Test_GetOpenapi() {
+  result=$($CurlCmd -H "$Header" "$url/openapi.json")
+  exit_code=$(GetExitCode "$result")
+  [[ $exit_code == 200 ]]
+}
+
+
+### Case: Post to composer
+function Test_PostToComposer() {
+  #
+  # Prepare a request to be sent to the composer API.
+  #
+
+  REQUEST_FILE="${WORKDIR}/request.json"
+  ARCH=$(uname -m)
+  SNAPSHOT_NAME=$(uuidgen)
+
+  case $(set +x; . /etc/os-release; echo "$ID-$VERSION_ID") in
+    "rhel-8.4")
+      DISTRO="rhel-84"
+    ;;
+    "rhel-8.2" | "rhel-8.3")
+      DISTRO="rhel-8"
+    ;;
+    "fedora-32")
+      DISTRO="fedora-32"
+    ;;
+    "fedora-33")
+      DISTRO="fedora-33"
+    ;;
+    "centos-8")
+      DISTRO="centos-8"
+    ;;
+  esac
+
+  cat > "$REQUEST_FILE" << EOF
+{
+  "distribution": "$DISTRO",
+  "customizations": {
+    "packages": [
+      "postgresql"
+    ]
+  },
+  "image_requests": [
+    {
+      "architecture": "$ARCH",
+      "image_type": "ami",
+      "repositories": $(jq ".\"$ARCH\"" /usr/share/tests/osbuild-composer/repositories/"$DISTRO".json),
+      "upload_requests": [
+        {
+          "type": "aws",
+          "options": {
+            "region": "${AWS_REGION}",
+            "s3": {
+              "access_key_id": "${AWS_ACCESS_KEY_ID}",
+              "secret_access_key": "${AWS_SECRET_ACCESS_KEY}",
+              "bucket": "${AWS_BUCKET}"
+            },
+            "ec2": {
+              "access_key_id": "${AWS_ACCESS_KEY_ID}",
+              "secret_access_key": "${AWS_SECRET_ACCESS_KEY}",
+              "snapshot_name": "${SNAPSHOT_NAME}",
+              "share_with_accounts": ["${AWS_API_TEST_SHARE_ACCOUNT}"]
+            }
+          }
+        }
+      ]
+    }
+  ]
+}
+
+EOF
+
+
+  #
+  # Send the request and wait for the job to finish.
+  #
+  # Separate `curl` and `jq` commands here, because piping them together hides
+  # the server's response in case of an error.
+  #
+
+  result=$($CurlCmd -H "$Header" -H 'Content-Type: application/json' --request POST --data @"$REQUEST_FILE" "$url/compose")
+  COMPOSE_ID=$(GetResponse "$result" | jq -r '.id')
```

Comment:
I guess we could verify that `$COMPOSE_ID` has a `uuid` format?

Reply:
Added checkpoint to verify the uuid format here.